### ISSUE TITLE
load an openapi spec from the web

### ISSUE THREAD

**Issue by pmcelhaney** [pmcelhaney/counterfact#104]
Currently the `generate` command expects the OpenAPI spec to be a file path. It should also work if passed an HTTP or HTTPS URL. 

In other words, `loadFile()` in [specification.js](https://github.com/pmcelhaney/counterfact/blob/main/src/typescript-generator/specification.js#L25-L40) should notice when the `path` is a URL and load the URL.

**Comment by KameronKales** [comment]
This ended up being a bit harder than I expected. Im documenting why here, and then we can decide what to do. 

The generate command takes a source and a destination. When the code was originally written there was no concept of fetching a file from a remote URL. The source as is, creates a full path from your local machine to the file in the repo. 

for example,  kameronkales/desktop/counterfact/petstore.yaml

However, if you try to feed a URL into the generate command you end up with something like this 

kameronkales/desktop/counterfact/https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v3.0/petstore.yaml

Internally counterfact tries to split the full path into multiple pieces. One of which being a basePath. That works fine when the file is local, but blows up when it builds a reference like this 

kameronkales/desktop/counterfact/https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v3.0

and 

petstore.yaml

After talking with Patrick, we're going to remove basePath as it exists today and refactor to use this. 

https://swagger.io/docs/specification/using-ref/

This will let us handle remote URIs and local the same. Im going to create a separate story to rip out basePath and get working with this new URI concept, then come back to this once it works.

**Comment by pmcelhaney** [comment]
Thanks! This will be a lot easier after #118
